core/src/main/java/uk/gov/gchq/koryphe/impl/predicate/If.java

```diff
@@ -0,0 +1,212 @@
+/*
+ * Copyright 2017 Crown Copyright
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package uk.gov.gchq.koryphe.impl.predicate;
+
+import com.fasterxml.jackson.annotation.JsonTypeInfo;
+import org.apache.commons.lang3.builder.EqualsBuilder;
+import org.apache.commons.lang3.builder.HashCodeBuilder;
+import org.apache.commons.lang3.builder.ToStringBuilder;
+
+import uk.gov.gchq.koryphe.ValidationResult;
+import uk.gov.gchq.koryphe.predicate.KoryphePredicate;
+import uk.gov.gchq.koryphe.signature.InputValidator;
+import uk.gov.gchq.koryphe.signature.Signature;
+import uk.gov.gchq.koryphe.tuple.Tuple;
+import uk.gov.gchq.koryphe.tuple.predicate.IntegerTupleAdaptedPredicate;
+
+import java.util.function.Predicate;
+
+/**
+ * An {@code If} is a {@link Predicate} that conditionally applies one of two predicates to a provided input.
+ *
+ * @param <I> the type of input to be validated
+ */
+@JsonTypeInfo(use = JsonTypeInfo.Id.CLASS, include = JsonTypeInfo.As.PROPERTY, property = "class")
+public class If<I> extends KoryphePredicate<I> {
+
+    private Boolean condition;
+
+    @JsonTypeInfo(use = JsonTypeInfo.Id.CLASS, include = JsonTypeInfo.As.PROPERTY, property = "class")
+    private Predicate<? super I> predicate;
```

Comment:
Would it be more user friendly to name these Predicates: "if", "then", "else"?

Reply:
Indeed it would, but unfortunately since "if" and "else" are keywords in Java, I've had to use alternatives for them - unless I used something like "ifCondition"?

Reply:
That's true, but that wouldn't stop us from being able to override these names in Jackson to make the JSON more friendly.

Reply:
That would be nice, but the python field names have to match the json field names, so it wouldn't work in python.